ompi/mpi/c/request_get_status_any.c.in

```diff
@@ -0,0 +1,136 @@
+ * Copyright (c) 2025      Advanced Micro Devices, Inc. All rights reserved
+ *
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+#include "ompi_config.h"
+#include <stdio.h>
+
+#include "ompi/mpi/c/bindings.h"
+#include "ompi/runtime/params.h"
+#include "ompi/communicator/communicator.h"
+#include "ompi/errhandler/errhandler.h"
+#include "ompi/request/request.h"
+#include "ompi/request/grequest.h"
+#include "ompi/memchecker.h"
+
+/* Non blocking status retrieval. Upon completion, the request(s) will
+ * not be freed (unlike the test function). A subsequent call to test, wait
+ * or free should be executed on the request(s).
+ */
+PROTOTYPE ERROR_CLASS request_get_status_any(INT count, REQUEST_CONST requests:count, INT_OUT indx,
+                                             INT_OUT flag,  STATUS_OUT status)
+{
+
+    MEMCHECKER(
+       int j;
+       for (j = 0; j< count;  j++) {
+           memchecker_request(&requests[j]);
+       }
+    );
+
+    if( MPI_PARAM_CHECK ) {
+        int rc = MPI_SUCCESS;
+
+        OMPI_ERR_INIT_FINALIZE(FUNC_NAME);
+        if (NULL == requests) {
+            rc = MPI_ERR_REQUEST;
+        } else {
+            /* Need to cast the const away, since the request argument for most
+            ** MPI functions invoking this functionality does not have the const
+            ** argument
+            */
+            if(!ompi_request_check_same_instance((MPI_Request *)requests, count) ) {
+                rc = MPI_ERR_REQUEST;
+            }
+        }
+        if ((NULL == flag) || (count < 0) || (NULL == indx) ) {
+            rc = MPI_ERR_ARG;
+        }
+        OMPI_ERRHANDLER_NOHANDLE_CHECK(rc, rc, FUNC_NAME);
+    }
+
+    if (OPAL_UNLIKELY(0 == count)) {
+        *flag = true;
+        *indx = MPI_UNDEFINED;
+        if (MPI_STATUS_IGNORE != status) {
+            OMPI_COPY_STATUS(status, ompi_status_empty, false);
+        }
+        return MPI_SUCCESS;
+    }
+
+    bool all_inactive;
+
+#if OPAL_ENABLE_PROGRESS_THREADS == 0
+    bool do_it_once = true;
+ recheck_request_status:
+#endif
+
+    opal_atomic_mb();
+    all_inactive = true;
+    int i;
+    for (i = 0; i < count; i++) {
+        if ( (requests[i] == MPI_REQUEST_NULL) || (requests[i]->req_state == OMPI_REQUEST_INACTIVE) ) {
+            continue;
+        }
+        if (requests[i]->req_complete ) {
+            *flag = true;
+            *indx = i;
+            /* If this is a generalized request, we *always* have to call
+               the query function to get the status (MPI-2:8.2), even if
+               the user passed STATUS_IGNORE. */
+            if (OMPI_REQUEST_GEN == requests[i]->req_type) {
+                ompi_grequest_invoke_query(requests[i], &requests[i]->req_status);
+            }
+            if (MPI_STATUS_IGNORE != status) {
+                OMPI_COPY_STATUS(status, requests[i]->req_status, false);
+            }
+            return MPI_SUCCESS;
+        } else {
```

Comment:
The `else` itself is unnecessary, it can be assumed from the rest of the checks. That's how it is done in the `some` function.